### DISCUSSION TITLE
Getting started with flux2 on an existing cluster

### DISCUSSION THREAD

**Discussion by ItielOlenick** [fluxcd/flux2#3431] [Q&A]
I would like to start using flux with an existing cluster.

Let's assume i have an application deployed using helm, and some additional resources applied with standard yaml files.

How can i bootstrap flux and have it takes over the existing configurations without replacing any resources?

Will having the helmRelease's name, namespace and version, coupled with the other resources matching the existing deployment's state be enough?

Would love to see an example or some documentation of onboarding to a preexisting cluster.

**Comment by stefanprodan** [comment]
> Will having the helmRelease's name, namespace and version, coupled with the other resources matching the existing deployment's state be enough?

Yes, for plain yamls you just place them in a dir and Flux will start managing. For Helm releases, you can use the Helm CLI to extract the values from the cluster then set them in the `HelmRelease.spec.values`, make sure that `HelmRelease.spec.releaseName` matches your current release.

Please see this example on how to structure your bootstrap repo: https://github.com/fluxcd/flux2-kustomize-helm-example
